deployer/deployer.py

```diff
@@ -297,23 +295,62 @@ def deploy_version_without_eureka(self, version, health_check, health_check_port
             }
         }
 
-        request = self.request("deployment/start", json.dumps(data))
-
-        time.sleep(30)
+        success = self.deploy_to_asgard(data, version)
+        if success:
+            self.resize_asg(asg_name=version)
 
-        self.resize_asg(asg_name=version)
+            good_deploy = self.wait_asg_ready(version=version,
+                                              health_check=health_check,
+                                              health_check_port=health_check_port)
 
-        good_deploy = self.wait_asg_ready(version=version,
-                                          health_check=health_check,
-                                          health_check_port=health_check_port)
+            if good_deploy:
+                if remove_old:
+                    self.remove_old_asg(new_asg_name=version)
+            else:
+                self.disable_asg(asg_name=version)
 
-        if good_deploy:
-            if remove_old:
-                self.remove_old_asg(new_asg_name=version)
         else:
-            self.disable_asg(asg_name=version)
+            raise Exception("Fucking asgard!")
+
+    def deploy_to_asgard(self, data, version):
+        wait_seconds_after_stack_start = 10
+        retries = 0
+        success = False
+        while not success and retries < 10:
```

Comment:
I think would be better to call it `tries`or `attempts` or even better IMO `retries_left`, and initialize to 10 and decrement on every loop.